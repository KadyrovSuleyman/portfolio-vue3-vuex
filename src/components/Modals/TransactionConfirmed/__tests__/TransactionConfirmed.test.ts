/* eslint-disable no-param-reassign */
import { mount, VueWrapper } from '@vue/test-utils';
import { createStore, Store } from 'vuex';
import Notification from '../TransactionConfirmed.vue';

jest.mock('../state.ts');

let store: Store<any>;
beforeEach(() => {
  store = createStore<any>({
    state: {
      isShown: false,
      hide: () => { store.state.isShown = false; },
    },
  });
});

let element: HTMLElement;
beforeEach(() => {
  element = document.createElement('div');
  element.className = 'app';
  document.body.appendChild(element);
});
afterEach(() => {
  document.body.outerHTML = '';
});

let wrapper: VueWrapper<any>;
afterEach(() => {
  wrapper.unmount();
});

it('transactionConfirmed renders', async () => {
  wrapper = mount(Notification, {
    global: { plugins: [store] },
    props: {
      target: '.app',
    },
  });

  expect(wrapper.findComponent('.transactionConfirmed-notification').exists()).toBeFalsy();

  store.state.isShown = true;
  await wrapper.vm.$nextTick();
  expect(store.state.isShown).toBeTruthy();
  expect(element.outerHTML).toMatchSnapshot();
});

it('watchs props changes', async () => {
  store.state.isShown = true;
  wrapper = mount(Notification, {
    global: { plugins: [store] },
    props: {
      target: '.app',
    },
  });
  expect(wrapper.findComponent('.transactionConfirmed-notification').classes())
    .toEqual(['transactionConfirmed-notification']);

  await wrapper.setProps({
    ...wrapper.props,
    mods: {
      selected: true,
    },
  });
  expect(wrapper.findComponent('.transactionConfirmed-notification').classes())
    .toEqual(['transactionConfirmed-notification', 'transactionConfirmed-notification__selected']);

  await wrapper.setProps({
    ...wrapper.props,
    mods: {
      selected: false,
    },
  });
  expect(wrapper.findComponent('.transactionConfirmed-notification').classes())
    .toEqual(['transactionConfirmed-notification']);
});

it('closes', async () => {
  wrapper = mount(Notification, {
    global: { plugins: [store] },
    props: {
      target: '.app',
    },
  });

  expect(wrapper.findComponent('.transactionConfirmed-notification').exists()).toBeFalsy();

  store.state.isShown = true;
  await wrapper.vm.$nextTick();
  expect(store.state.isShown).toBeTruthy();
  expect(wrapper.findComponent('.transactionConfirmed-notification').exists()).toBeTruthy();

  wrapper.findComponent('.notification-button').trigger('click');
  await wrapper.vm.$nextTick();
  expect(store.state.isShown).toBeFalsy();
  expect(wrapper.findComponent('.transactionConfirmed-notification').exists()).toBeFalsy();
});

it('notification clicks', async () => {
  wrapper = mount(Notification, {
    global: { plugins: [store] },
    props: {
      target: '.app',
    },
  });

  store.replaceState({
    hide: jest.fn(),
  });

  expect(wrapper.findComponent('.transactionConfirmed-notification').exists()).toBeFalsy();

  store.state.isShown = true;
  await wrapper.vm.$nextTick();
  expect(store.state.isShown).toBeTruthy();
  expect(wrapper.findComponent('.transactionConfirmed-notification').exists()).toBeTruthy();
  expect(store.state.hide).toBeCalledTimes(0);

  await wrapper.findComponent('.transactionConfirmed-button').trigger('click');
  expect(store.state.hide).toBeCalledTimes(1);

  await wrapper.findComponent('.transactionConfirmed-button').trigger('click');
  expect(store.state.hide).toBeCalledTimes(2);
});
