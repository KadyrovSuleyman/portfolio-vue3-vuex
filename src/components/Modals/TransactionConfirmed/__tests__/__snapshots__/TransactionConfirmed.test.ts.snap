// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transactionConfirmed renders 1`] = `
<div class="app">
  <transition-stub>
    <div class="transactionConfirmed-notification"><button class="transactionConfirmed-button"><span class="transactionConfirmed-span"> Transaction confirmed! Click here to see it </span></button><button class="notification-button notification-button__icon_cross_white"></button></div>
  </transition-stub>
</div>
`;
